pkg/trace/obfuscate/json.go

```diff
@@ -25,27 +27,47 @@ func (o *Obfuscator) obfuscateJSON(span *pb.Span, tag string, obfuscator *jsonOb
 	// as much of it as we could. It is safe to accept the output, even if partial.
 }
 
+type valueTransformer func(string) string
+
 type jsonObfuscator struct {
-	keepers map[string]bool // these keys will not be obfuscated
+	keepKeys      map[string]bool // these keys will not be obfuscated
+	transformKeys map[string]bool // these keys pass through the transformer
+	transformer   valueTransformer
 
 	scan     *scanner // scanner
 	closures []bool   // closure stack, true if object (e.g. {[{ => []bool{true, false, true})
 	key      bool     // true if scanning a key
 
-	wiped     bool // true if obfuscation string (`"?"`) was already written for current value
-	keeping   bool // true if not obfuscating
-	keepDepth int  // the depth at which we've stopped obfuscating
+	wiped             bool // true if obfuscation string (`"?"`) was already written for current value
+	keeping           bool // true if not obfuscating
+	transformingValue bool // true if collecting the next literal for transformation
+	keepDepth         int  // the depth at which we've stopped obfuscating
 }
 
-func newJSONObfuscator(cfg *config.JSONObfuscationConfig) *jsonObfuscator {
+func (o *Obfuscator) newJSONObfuscator(cfg *config.JSONObfuscationConfig) *jsonObfuscator {
 	keepValue := make(map[string]bool, len(cfg.KeepValues))
 	for _, v := range cfg.KeepValues {
 		keepValue[v] = true
 	}
+	var transformValues map[string]bool
+	var transformer valueTransformer
+	if cfg.TransformerType == "obfuscate_sql" {
+		transformer = o.ObfuscateSQLStringSafe
+	} else {
+		log.Warnf("unknown JSON value transformer type %s", cfg.TransformerType)
+	}
+	if transformer != nil {
+		transformValues = make(map[string]bool, len(cfg.TransformValues))
```

Comment:
Wouldn't it be easier for `cfg.TransformValues` to be a map from the git-go?

Reply:
We could only if we make it a map in yaml, which I don't think is a good idea because this kind of setting seems nicer to have as a list in yaml. 

Better the user see 
`keys: ['a', 'b', 'c']`

than 

```
keys:
    a: true
    b: true
    c: true
```